cmd/account/mgmt/account-assign_test.go

```diff
@@ -1,18 +1,89 @@
 package mgmt
 
 import (
+	"bytes"
 	"fmt"
 	"math/rand"
+	"os"
 	"testing"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/organizations"
 	"github.com/golang/mock/gomock"
+	"github.com/openshift/osdctl/internal/utils/globalflags"
+	awsprovider "github.com/openshift/osdctl/pkg/provider/aws"
 	"github.com/openshift/osdctl/pkg/provider/aws/mock"
 
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
 )
 
+func TestAccountAassignCommand(t *testing.T) {
+
+	mocks := setupDefaultMocks(t, []runtime.Object{})
```

Comment:
I see that you call out having to change up the command instantiation to test this, but I'm wondering if there was any reason why you couldn't just do something like:

```
o := accountAssignOptions{}
o.awsClient = mockAwsClient
o.username = "myuser"
o.accountid = "12345"

err := o.run()
```

instead of running this big integration-test like unit test?

Reply:
Yup, I could have done exactly that! Honestly this mainly just stemmed from curiosity to see if I could find an interesting way to run integration tests, so I'm happy to create a `o.run()` unit test too if you'd like?

Reply:
I've moved the existing test into an integration test file, and created a unit test for isOwned

> Maybe use that to test raw input handling but not specifically the functionality of the operations in the command?
If we're just testing the input and not the command operations then we're kinda just writing unit tests for the Cobra command library then?

Reply:
Yes and no.  What I imagine is in conditions like where we need to specify one flag OR another, we would have a few test cases that are testing the input validation function we have:

1. Neither flag is passed - test that command errors
2 & 3. one and only one of each flag is passed - test that command succeeds
4. Both flags passed - test that command errors

Though I guess the way we structure this we could handle this with a function that can also get unit tested as well, instead of doing a full integration test.

So yeah, just unit testing here would work.  I'm not going to ask you to do _more_ work for little to no real gain :)